functions/src/index.ts

```diff
@@ -242,13 +328,52 @@ exports.calculateAqi = functions.pubsub
           nowCastPm25: nowcastPm25.reading,
           aqi: aqi,
         };
+        aqiBufferData = {
+          aqi: aqi,
+          timestamp: FieldValue.serverTimestamp(),
+        };
       }
-    }
+      // Send AQI reading to current-readings to be displayed on the map
+      await firestore.collection('current-reading').doc('pm25').set({
+        lastUpdated: FieldValue.serverTimestamp(),
+        data: currentData,
+      });
 
-    await firestore.collection('current-reading').doc('pm25').set({
-      lastUpdated: FieldValue.serverTimestamp(),
-      data: currentData,
-    });
+      /// / -------- Update the AQI Circular Buffer -------- ////
+
+      // Document reference for current sensor
+      const sensorDocRef = firestore.collection('/sensors').doc(docId);
+
+      await sensorDocRef.get().then(doc => {
```

Comment:
Similar confusion on the combo `async`/`await` and `.then()`

Reply:
Same response, will recreate bug (or bug will be solved).